### PR TITLE
UXD-304 Button Separated Variations

### PR DESCRIPTION
**Pull request by KDarcey**
### Purpose 🚀
To separate the different button variations to distinguish the button options (Link, Close, Icon, etc).

### Notes ✏️
- All variations are currently still displayed in one location on storybook (may be separated later).

### Updates 📦
- [ ] MAJOR (breaking) change to _these packages_
- [ ] MINOR (backward compatible) change to _these packages_
- [x] PATCH (bug fix) change to _these packages_

### Storybook 📕
http://storybooks.highbond-s3.com/paprika/UXD-304-Button-split-variations

### Screenshots 📸
N/A

### References 🔗
https://aclgrc.atlassian.net/browse/UXD-304

### CODE REVIEW

**Comment by nahumzs** [comment]
would this be part of a bigger part to include the `mdx` doc file for the buttons? 🙇

File: packages/Button/stories/variations/Variations.js [hunk 1]
```diff
@@ -1,324 +1,23 @@
...
-};
+import ButtonVariations from "./ButtonVariations";
+import LinkButtonVariations from "./LinkButtonVariations";
+import IconButtonVariations from "./IconButtonVariations";
+import CloseButtonVariations from "./CloseButtonVariations";
 
 export default function Variations() {
   return (
     <ButtonStory>
       <Tagline>So many different kinds of Buttons!</Tagline>
```
Comment (mikrotron) [inline packages/Button/stories/variations/Variations.js:12]: Let's add a `<Heading level={1} displayLevel={2} isLight>Variations of <Button></Heading>` here.

Comment (mikrotron) [reply packages/Button/stories/variations/Variations.js:12]: Or maybe it should go in the `ButtonVariations.js` file.  In that case we can move this `<Tagline>` too (or just get rid of it).

File: packages/Button/stories/variations/LinkButtonVariations.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import Button from "../../src/Button";
+
+function clickHandler() {
+  action("Clicked a button")();
+}
+
+export default function LinkButtonVariations() {
+  return (
+    <>
+      <p>
```
Comment (mikrotron) [inline packages/Button/stories/variations/LinkButtonVariations.js:14]: Let's add a "Variations of <Button.Link>" heading here.  And for the others. You get the idea.

**Review by mikrotron — APPROVED** [review]
llgtm 91

**Pull request merged**